crates/aptos-rosetta/src/common.rs

```diff
@@ -203,6 +199,116 @@ pub async fn get_block_index_from_request(
     })
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct BlockHash {
+    chain_id: ChainId,
+    block_height: u64,
+}
+
+impl BlockHash {
+    pub fn block_height(&self, chain_id: ChainId) -> ApiResult<u64> {
+        if chain_id != self.chain_id {
+            Err(ApiError::InvalidInput(Some(format!(
+                "Invalid chain id in block hash {} expected {}",
+                self.chain_id, chain_id
+            ))))
+        } else {
+            Ok(self.block_height)
+        }
+    }
+}
+
+impl From<(ChainId, u64)> for BlockHash {
+    fn from((chain_id, block_height): (ChainId, u64)) -> Self {
+        BlockHash {
+            chain_id,
+            block_height,
+        }
+    }
+}
+
+impl FromStr for BlockHash {
+    type Err = ApiError;
+
+    fn from_str(str: &str) -> Result<Self, Self::Err> {
+        let error = ApiError::InvalidInput(Some(format!("Invalid block hash {}", str)));
```

Comment:
Can we not do this, and just map err in each with a more specific error, e.g. `"Invalid block hash {}: Invalid chain ID", str`. There are 3 different error cases, let's have a specific message for each, otherwise we won't know what's wrong.